### PR TITLE
refactor: update snippets/embedded example comments in material docs md to reflect new API (autocomplete, datepicker, and menu)

### PR DESCRIPTION
**Pull request by annieyw**
Remove static code snippets in material docs and replace them with embedded examples in compact view.
Add regions in component examples.

### CODE REVIEW

File: src/material/autocomplete/autocomplete.md [hunk 1]
```diff
@@ -2,51 +2,33 @@ The autocomplete is a normal text input enhanced by a panel of suggested options
...
-  <mat-option *ngFor="let option of options" [value]="option">
-    {{ option }}
-  </mat-option>
-</mat-autocomplete>
-```
+<!-- example({"example":"autocomplete-simple",
+              "file":"autocomplete-simple-example.html", 
+              "lines":[2, 8]}) -->
 
 Now we'll need to link the text input to its panel. We can do this by exporting the autocomplete
```
Comment (mmalerba) [inline src/material/autocomplete/autocomplete.md:26]: I think you can just remove this last step, now that you covered it above

File: src/material/datepicker/datepicker.md [hunk 2]
```diff
@@ -8,29 +8,17 @@ the calendar. It is made up of several components and directives that work toget
 A datepicker is composed of a text input and a calendar pop-up, connected via the `matDatepicker`
 property on the text input.
 
-```html
-<input [matDatepicker]="myDatepicker">
-<mat-datepicker #myDatepicker></mat-datepicker>
-```
+An optional datepicker toggle button is available.
```
Comment (mmalerba) [inline src/material/datepicker/datepicker.md:11]: "There is also an optional datepicker toggle button that gives the user an easy way to open the datepicker pop-up"

File: src/material/autocomplete/autocomplete.md [hunk 3]
```diff
@@ -2,51 +2,33 @@ The autocomplete is a normal text input enhanced by a panel of suggested options
 
 ### Simple autocomplete
 
-Start by adding a regular `matInput` to your template. Let's assume you're using the `formControl`
-directive from `ReactiveFormsModule` to track the value of the input.
+Start by creating the autocomplete panel and the options displayed inside it. Each option should be
+defined by a `mat-option` tag. Set each option's value property to whatever you'd like the value
+of the text input to be upon that option's selection.
```
Comment (atscott) [inline src/material/autocomplete/autocomplete.md:7]:
```suggestion
of the text input to be when that option is selected.
```
Optional revision.

File: src/material/autocomplete/autocomplete.md [hunk 4]
```diff
@@ -2,51 +2,33 @@ The autocomplete is a normal text input enhanced by a panel of suggested options
...
+
+<!-- example({"example":"autocomplete-simple",
+              "file":"autocomplete-simple-example.html", 
+              "lines":[8, 13]}) -->
+
+Next, create the input and set the matAutocomplete input to refer to the template reference we put on the autocomplete.
```
Comment (atscott) [inline src/material/autocomplete/autocomplete.md:13]:
```suggestion
Next, create the input and set the `matAutocomplete` input to refer to the template reference we assigned to the autocomplete.
```

**Pull request merged**